Include/cpython/pystate.h

```diff
@@ -176,8 +176,28 @@ struct _ts {
     PyObject **datastack_limit;
     /* XXX signal handlers should also be here */
 
+    // This must be last.
+    struct {
+        int initialized;
+    } _preallocated;
 };
 
+#ifdef Py_BUILD_CORE
+#define _PyThreadState_INIT \
+    { \
+        ._preallocated = { \
+            .initialized = 1, \
+        }, \
+    }
+
+static inline void
+_PyThreadState_reset(PyThreadState *tstate)
```

Comment:
I don't think this is safe.
What happens a running thread is reset?
Could you remove this?

Reply:
It isn't part of the public C-API and is only used for initializing "new" threads states.  I'll move it into pystate.c so there's no confusion.

Reply:
Actually, I've dropped it.